src/main/java/nl/tudelft/ewi/devhub/server/web/resources/WarningResource.java

```diff
@@ -0,0 +1,115 @@
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+
+/**
+ * Created by LC on 29/04/15.
+ */
+
+@Path("/courses/{courseCode}/groups/{groupId}/warnings")
+public class WarningResource extends Resource {
+
+    private final Group group;
+
+    private final User currentUser;
+
+    private final PullRequests pullRequests;
+
+    private final Deliveries deliveries;
+
+    private final TemplateEngine templateEngine;
+
+    private final Repository repository;
+
+    @Inject
+    public WarningResource( @Named("current.group")  Group group, @Named("current.user")User currentUser,
+                            PullRequests pullRequests, Deliveries deliveries, TemplateEngine templateEngine,
+                            @Named("current.repository") Repository repository) {
+        this.group = group;
+        this.currentUser = currentUser;
+        this.pullRequests = pullRequests;
+        this.deliveries = deliveries;
+        this.templateEngine = templateEngine;
+        this.repository = repository;
+    }
+
+
+    @GET
+    @Produces(MediaType.TEXT_HTML + Resource.UTF8_CHARSET)
+    public Response showProjectSetupPageStep1(@Context HttpServletRequest request) throws IOException, GitClientException {
+        Map<String, Object> parameters = Maps.newHashMap();
+        parameters.put("user", currentUser);
+        parameters.put("group",group);
+        parameters.put("course",group.getCourse());
+        parameters.put("repository",repository);
+
+        List<Locale> locales = Collections.list(request.getLocales());
+        return display(templateEngine.process("warnings-view.ftl", locales, parameters));
+    }
+
+
+    /**
+     * Api call to get the pullrequest status of a group for the current assignment.
+     * @return true if there has been made use of pull-requests for this assignment false otherwise.
+     *
+     */
+    @GET
+    @Path("/pull-usage")
+    @Produces(MediaType.APPLICATION_JSON)
+    public PullRequestWarningResponse getPullRequestUsageStatus(){
+        Delivery lastDelivery = deliveries.getLastDelivery(group);
+        Stream<PullRequest> pullRequestStream = pullRequests.findAllPullRequests(group).stream();
+
+        if(lastDelivery != null) {
+            Date lastDeliveryDate = lastDelivery.getCreated();
+            pullRequestStream = pullRequestStream.filter((pr) -> lastDeliveryDate.before(mapDatePullRequest(pr)));
+        }
+        return findStatus(pullRequestStream.collect(Collectors.toList()));
+    }
+
+    public static PullRequestWarningResponse findStatus(List<PullRequest> pullRequests){
+        if(pullRequests.stream().filter(PullRequest::isClosed).filter(PullRequest::isMerged).findAny().isPresent()) {
```

Comment:
I think only filtering only merged pull requests will do. (They will be closed anyway, maybe we could allow to reopen them later on, but even then, the PR boolean should be the only relevant flag for this check)